src/Rules/Operators/InvalidBinaryOperationRule.php

```diff
@@ -113,6 +116,30 @@ public function processNode(Node $node, Scope $scope): array
 			];
 		}
 
+		if ($this->phpVersion->deprecatesBitwiseOperationsWithFloat()) {
+			if (
+				$node instanceof Node\Expr\BinaryOp\ShiftRight
+				|| $node instanceof Node\Expr\BinaryOp\ShiftLeft
+				|| $node instanceof Node\Expr\BinaryOp\BitwiseOr
+				|| $node instanceof Node\Expr\BinaryOp\BitwiseAnd
+				|| $node instanceof Node\Expr\BinaryOp\BitwiseXor
+			) {
+				$leftType = $scope->getType($node->left);
+				$rightType = $scope->getType($node->right);
+				$floatType = new FloatType();
+				if ($floatType->isSuperTypeOf($leftType)->yes() || $floatType->isSuperTypeOf($rightType)->yes()) {
```

Comment:
not sure whether these checks should actually be `maybe()` to account for union types etc?